save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/internal/AgentsController.kt

```diff
@@ -18,15 +21,52 @@ import org.springframework.web.bind.annotation.RequestMapping
 import org.springframework.web.bind.annotation.RequestParam
 import org.springframework.web.bind.annotation.RestController
 import org.springframework.web.server.ResponseStatusException
+import reactor.core.publisher.Mono
 
 /**
  * Controller to manipulate with Agent related data
  */
 @RestController
 @RequestMapping("/internal")
-class AgentsController(private val agentStatusRepository: AgentStatusRepository,
-                       private val agentRepository: AgentRepository,
+class AgentsController(
+    private val agentStatusRepository: AgentStatusRepository,
+    private val agentRepository: AgentRepository,
+    private val configProperties: ConfigProperties,
 ) {
+    /**
+     * @param containerId [Agent.containerId]
+     * @return [Mono] with [AgentInitConfig]
+     */
+    @GetMapping("/agents/get-init-config")
+    fun getInitConfig(
+        @RequestParam containerId: String,
+    ): Mono<AgentInitConfig> = blockingToMono {
+        agentRepository.findByContainerId(containerId)
+    }
+        .switchIfEmptyToNotFound {
+            "Not found agent with container id $containerId"
+        }
+        .map {
+            it.execution
```

Comment:
execution is not available here due to `fetch = FetchType.LAZY`, need to try `Transaction(readonly)`

Reply:
didn't help, changed to `FetchType.EAGER` as a simple solution